gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureGeneratorTask.java

```diff
@@ -92,17 +102,17 @@ public void compileFiles() {
 
             try {
                 FileUtils.deleteDirectory(thisOutputDirectory);
+                Files.createDirectories(thisOutputDirectory.toPath());
             } catch (IOException e) {
                 throw new UncheckedIOException(e);
             }
 
-            getProject().mkdir(thisOutputDirectory);
-
             List<String> generateCommand =
                     ImmutableList.of("generate", file.getAbsolutePath(), thisOutputDirectory.getAbsolutePath());
 
             GradleExecUtils.exec(
-                    getProject(),
+                    getExecOperations(),
+                    getBuildServiceRegistry(),
```

Comment:
I think a nicer to structure this would be to make a `GradleExecUtils` a managed type, where `ExecOperations` and `BuildServiceRegistry` are injected into it (like they currently are this task). This means we can just add it as `@Nested` and don't need to inject all it's dependencies (ie `ExecOperations` and `BuildServiceRegistry`) into each task.